lib/Elevate/Components/SSH.pm

```diff
@@ -0,0 +1,57 @@
+package Elevate::Components::SSH;
+
+=encoding utf-8
+
+=head1 NAME
+
+Elevate::Components::SSH
+
+Ensure that the sshd config file has 'PermitRootLogin' set to 'yes'
+if it is not set.
+
+=cut
+
+use cPstrict;
+
+use Elevate::Constants ();
+
+use Cwd           ();
+use File::Slurper ();
+use Log::Log4perl qw(:easy);
+
+use parent qw{Elevate::Components::Base};
+
+sub pre_leapp ($self) {
+
+    my $sshd_config = q[/etc/ssh/sshd_config];
+
+    my $setup = eval { File::Slurper::read_binary($sshd_config) } // '';
+
+    # PermitRootLogin is explicitly set, no need for changes
+    return if ( $setup =~ m{^\s*PermitRootLogin\b}m );
+
+    my $line_to_append = "PermitRootLogin yes\n";
+
+    # Prefix with newline if file does not end with newline
+    if ( $setup !~ m{\n$} && length $setup ) {
+        $line_to_append = "\n" . $line_to_append;
+    }
+
+    open( my $fh, '>>', $sshd_config ) or do {
+        WARN("Could not open $sshd_config to update PermitRootLogin ($!). This will default to \"no\" after the upgrade.");
```

Comment:
Yeah, that makes sense.  I'll switch it to FATAL().  If we're running as root & can't open the sshd config file for writing, then something weird is going on.  I'll also throw a fatal error if File::Slurp fails to read the sshd config file; because that should definitely work as well.

Reply:
please use File::Slurper. it will die if it fails to read/write anyway.